### PR TITLE
python310Packages.werkzeug: 2.2.3 -> 2.3.6

### PR DESCRIPTION
**Pull request by blaggacao**
- python310Packages.werkzeug: 2.2.3 -> 2.3.6
- python310Packages.httpbin: disable failing tests with werkzeug 2.3.6

###### Description of changes

Changelog: https://werkzeug.palletsprojects.com/en/2.3.x/changes/

Mainly motivated by: https://github.com/pallets/werkzeug/issues/2603

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/werkzeug/default.nix [hunk 1]
```diff
@@ -15,17 +16,21 @@
...
   src = fetchPypi {
     pname = "Werkzeug";
     inherit version;
-    hash = "sha256-LhzMlBfU2jWLnebxdOOsCUOR6h1PvvLWZ4ZdgZ39Cv4=";
+    hash = "sha256-mMd03y+RsFVQB4iR3uXw6wy3l6Uix1eiRSuc7lsgIzA=";
   };
 
+  nativeBuildInputs = [
+    setuptools
+  ];
```
Comment (tjni) [inline pkgs/development/python-modules/werkzeug/default.nix:32]: Can you also add wheel to this? It's implicitly added right now by our Python build hooks, but I'm working on changing this.

File: pkgs/development/python-modules/werkzeug/default.nix [hunk 2]
```diff
@@ -15,17 +16,21 @@
 
 buildPythonPackage rec {
   pname = "werkzeug";
-  version = "2.2.3";
-  format = "setuptools";
+  version = "2.3.6";
+  format = "pyproject";
 
   disabled = pythonOlder "3.7";
```
Comment (tjni) [inline pkgs/development/python-modules/werkzeug/default.nix:22]: This version of werkzeug dropped support for Python 3.7.

File: pkgs/development/python-modules/httpbin/default.nix [hunk 3]
```diff
@@ -62,6 +62,11 @@ buildPythonPackage rec {
     "test_redirect_to_post"
     "test_relative_redirect_n_equals_to_1"
     "test_relative_redirect_n_higher_than_1"
+    # break with new werkzeug
+    # disable until PSF fork catches up: https://github.com/psf/httpbin
+    "test_base64"
+    "test_digest_auth"
+    "test_digest_auth_wrong_pass"
```
Comment (tjni) [inline pkgs/development/python-modules/httpbin/default.nix:69]: Would you mind trying if these two patches fix these test issues?

https://github.com/psf/httpbin/commit/b4e9808a628d1460346ec8790d810586f03bb4ba.patch

https://github.com/psf/httpbin/commit/7facfd8ea1983a4dd8e759d11155e81a682c16a1.patch

**Review by tjni — COMMENTED** [review]
Looks good!

File: pkgs/development/python-modules/httpbin/default.nix (hunk 3 above)
Comment (blaggacao) [reply pkgs/development/python-modules/httpbin/default.nix]: Indeed, it worked: https://github.com/NixOS/nixpkgs/compare/77adbeb69e1da83bc8f87e05667345fbbc00aa77..c17fc82201157b2f00e5520052ac8de30062f78c

File: pkgs/development/python-modules/werkzeug/default.nix (hunk 2 above)
Comment (blaggacao) [reply pkgs/development/python-modules/werkzeug/default.nix]: fixed in [force-push](https://github.com/NixOS/nixpkgs/compare/77adbeb69e1da83bc8f87e05667345fbbc00aa77..c17fc82201157b2f00e5520052ac8de30062f78c)

File: pkgs/development/python-modules/werkzeug/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/development/python-modules/werkzeug/default.nix]: fixed in [force-push](https://github.com/NixOS/nixpkgs/compare/77adbeb69e1da83bc8f87e05667345fbbc00aa77..c17fc82201157b2f00e5520052ac8de30062f78c)

**Review by tjni — APPROVED** [review]
LGTM, thank you for the changes!

**Comment by wegank** [comment]
Merge conflict, please rebase.

**Pull request closed without merging**